shell/platform/windows/public/flutter_windows.h

```diff
@@ -165,6 +174,22 @@ FLUTTER_EXPORT bool FlutterDesktopEngineDestroy(FlutterDesktopEngineRef engine);
 FLUTTER_EXPORT bool FlutterDesktopEngineRun(FlutterDesktopEngineRef engine,
                                             const char* entry_point);
 
+// Creates a view for the given engine.
+//
+// The |engine| will be started if it is not already running.
+//
+// The caller owns the returned reference, and is responsible for calling
+// |FlutterDesktopViewControllerDestroy|. Returns a null pointer in the event of
+// an error.
+//
+// Unlike |FlutterDesktopViewControllerCreate|, this does *not* take ownership
+// of |engine| and |FlutterDesktopEngineDestroy| must be called to destroy
+// the engine.
+FLUTTER_EXPORT FlutterDesktopViewControllerRef
+FlutterDesktopEngineCreateViewController(
+    FlutterDesktopEngineRef engine,
+    const FlutterDesktopViewControllerProperties* properties);
```

Comment:
FYI, we were holding off on making the multi-window APIs public until the Windows embedder properly supports multi-window.

Known problems with Window's multi-window support: https://flutter.dev/go/multi-window-status#heading=h.i5rwjhitgcmu

1. Cursor support
2. Keyboard support
3. Text input support
4. App lifecycle across multiple windows
5. Accessibility support
6. Plugins updated

Could we keep all these APIs non-public until we're sure these are production ready?

Reply:
I hope it can be public before the above 6 items are ready, so that more people can prototype this function; 
it is enough for flutter to just warn that it is not production ready;

Reply:
> I hope it can be public before the above 6 items are ready, so that more people can prototype this function; it is enough for flutter to just warn that it is not production ready;

The API is "pubternal"; advanced users can opt-in to using it (see https://github.com/flutter/flutter/issues/143767#issuecomment-2050741185). I think that's a good compromise as it prevents folks from unintentionally using an API that we know is broken.